tests/core/test_solve.py

```diff
@@ -363,6 +363,153 @@ def test_prune_1(tmpdir):
+            'channel-1::llvmpy-0.11.2-py27_0',
+            'channel-1::meta-0.4.2.dev-py27_0',
+            'channel-1::mkl-service-1.0.0-py27_p0',
+            'channel-1::numpy-1.7.1-py27_p0',
+            'channel-1::numba-0.8.1-np17py27_0',
+            'channel-1::numexpr-2.1-np17py27_p0',
+            'channel-1::scipy-0.12.0-np17py27_p0',
+            'channel-1::numbapro-0.11.0-np17py27_p0',
+            'channel-1::scikit-learn-0.13.1-np17py27_p0',
+            'channel-1::mkl-11.0-np17py27_p0',
+            'channel-1::accelerate-1.1.0-np17py27_p0',
+        ))
+        assert convert_to_dist_str(final_state_1) == order
+
+    new_environment_specs = MatchSpec("python=2.7.3"),
+
+    with get_solver(tmpdir, new_environment_specs, prefix_records=final_state_1,
+                    history_specs=specs) as solver:
+        final_state_1 = solver.solve_final_state(prune=True)
+        pprint(convert_to_dist_str(final_state_1))
+        order = add_subdir_to_iter((
+            'channel-1::openssl-1.0.1c-0',
+            'channel-1::readline-6.2-0',
+            'channel-1::sqlite-3.7.13-0',
+            'channel-1::system-5.8-1',
+            'channel-1::tk-8.5.13-0',
+            'channel-1::zlib-1.2.7-0',
+            'channel-1::python-2.7.3-7',
+        ))
+        assert convert_to_dist_str(final_state_1) == order
+
+
+def test_update_prune_3(tmpdir):
+    """Ensure that update with prune is not removing packages that are still
+    needed by remaining specs.
+    """
+    specs = MatchSpec("numpy"), MatchSpec("python=2.7.3"), MatchSpec("accelerate"),
+
+    with get_solver(tmpdir, specs) as solver:
+        final_state_1 = solver.solve_final_state()
+        pprint(convert_to_dist_str(final_state_1))
+        order = add_subdir_to_iter((
+            'channel-1::libnvvm-1.0-p0',
+            'channel-1::mkl-rt-11.0-p0',
+            'channel-1::openssl-1.0.1c-0',
+            'channel-1::readline-6.2-0',
+            'channel-1::sqlite-3.7.13-0',
+            'channel-1::system-5.8-1',
+            'channel-1::tk-8.5.13-0',
+            'channel-1::zlib-1.2.7-0',
+            'channel-1::llvm-3.2-0',
+            'channel-1::python-2.7.3-7',
+            'channel-1::bitarray-0.8.1-py27_0',
+            'channel-1::llvmpy-0.11.2-py27_0',
+            'channel-1::meta-0.4.2.dev-py27_0',
+            'channel-1::mkl-service-1.0.0-py27_p0',
+            'channel-1::numpy-1.7.1-py27_p0',
+            'channel-1::numba-0.8.1-np17py27_0',
+            'channel-1::numexpr-2.1-np17py27_p0',
+            'channel-1::scipy-0.12.0-np17py27_p0',
+            'channel-1::numbapro-0.11.0-np17py27_p0',
+            'channel-1::scikit-learn-0.13.1-np17py27_p0',
+            'channel-1::mkl-11.0-np17py27_p0',
+            'channel-1::accelerate-1.1.0-np17py27_p0',
+        ))
+        assert convert_to_dist_str(final_state_1) == order
+
+    new_environment_specs = MatchSpec("numpy"), MatchSpec("python=2.7.3"),
```

Comment:
I think here would be a good place to comment what's the relation between the new specs and the test intent according to the docstring (same applies to the other tests).